src/etc/inc/pkg-utils.inc

```diff
@@ -359,21 +360,35 @@ function get_pkg_info($pkgs = 'all', $only_local = false) {
 
 	/* Do not run remote operations if pkg has a lock */
 	if (is_subsystem_dirty('pkg')) {
-		$only_local = true;
+		$local_only = true;
 		$lock = false;
 	} else {
 		$lock = true;
 	}
 
 	$extra_param = "";
-	if ($only_local) {
+	if ($local_only) {
 		$extra_param = "-U ";
 	}
 
 	if ($lock) {
 		mark_subsystem_dirty('pkg');
 	}
-	$rc = pkg_exec("search {$extra_param}-R --raw-format json-compact " . $pkgs, $out, $err);
+	if (!$installed_only) {
+		// repo catalog search (either remote or local_only)
+		$rc = pkg_exec("search {$extra_param}-R --raw-format json-compact " . $pkgs, $out, $err);
+	}
+	if (($rc != 0 && $local_only) || $installed_only) {
+		/* use pkg info if (1) installed pkg search or (2) local catalog copy search requested + failed.
+		
+		   The local repo catalog copy may be cleared if a previous call to pkg search couldn't get the remote repo catalog.
+		   
+		   If the calling code would have accepted local copy info (which isn't assumed up to date) then it makes sense to
+		   fall back on pkg info to at least return the known info about installed pkgs (pkg info should still work), instead
+		   of failing and returning no info at all. For example, this enables offline view + management of installed pkgs. */
```

Comment:
Since there are another change to do please use following format for multi line comments and try to keep it in 80 columns:
```
/*
 * multiline comment
 * should be formatted like this
 */
```